types/proposal_test.go

```diff
@@ -190,3 +206,143 @@ func TestProposalProtoBuf(t *testing.T) {
 		}
 	}
 }
+
+func TestTxMetaData_RoundTrip(t *testing.T) {
+	tests := []struct {
+		name string
+		data *TxMetaData
+	}{
+		{
+			"valid metadata",
+			&TxMetaData{Hash: cmtrand.Bytes(tmhash.Size), Start: 0, End: 10},
+		},
+		{
+			"empty hash",
+			&TxMetaData{Hash: []byte{}, Start: 5, End: 10},
+		},
+		{
+			"same start and end",
+			&TxMetaData{Hash: cmtrand.Bytes(tmhash.Size), Start: 7, End: 7},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			proto := tt.data.ToProto()
+			roundTripped := TxMetaDataFromProto(proto)
+			require.Equal(t, tt.data, roundTripped)
+		})
+	}
+}
+
+func TestTxMetaData_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name    string
+		data    *TxMetaData
+		expects error
+	}{
+		{
+			"valid metadata",
+			&TxMetaData{Hash: cmtrand.Bytes(tmhash.Size), Start: 0, End: 10},
+			nil,
+		},
+		{
+			"invalid start > end",
+			&TxMetaData{Hash: cmtrand.Bytes(tmhash.Size), Start: 10, End: 5},
+			errors.New("TxMetaData: Start > End"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.data.ValidateBasic()
+			if tt.expects != nil {
+				assert.EqualError(t, err, tt.expects.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestCompactBlock_RoundTrip(t *testing.T) {
+	tests := []struct {
+		name string
+		data *CompactBlock
+	}{
+		{
+			"valid block",
+			&CompactBlock{
+				Height: 1, Round: 1, BpHash: cmtrand.Bytes(tmhash.Size),
+				Blobs:     []*TxMetaData{{Hash: cmtrand.Bytes(tmhash.Size), Start: 0, End: 10}},
+				Signature: cmtrand.Bytes(MaxSignatureSize),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			proto := tt.data.ToProto()
+			// Assuming CompactBlockFromProto exists
+			roundTripped, err := CompactBlockFromProto(proto)
+			require.NoError(t, err)
+			require.Equal(t, tt.data, roundTripped)
+		})
+	}
+}
+
+func TestCompactBlock_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name    string
+		data    *CompactBlock
+		expects error
+	}{
+		{
+			"valid block",
+			&CompactBlock{
+				Height: 1, Round: 1, BpHash: cmtrand.Bytes(tmhash.Size),
+				Blobs:     []*TxMetaData{{Hash: cmtrand.Bytes(tmhash.Size), Start: 0, End: 10}},
+				Signature: cmtrand.Bytes(MaxSignatureSize),
+			},
+			nil,
+		},
+		{
+			"negative height",
+			&CompactBlock{
+				Height: -1, Round: 1, BpHash: cmtrand.Bytes(tmhash.Size),
+				Blobs:     []*TxMetaData{{Hash: cmtrand.Bytes(tmhash.Size), Start: 0, End: 10}},
+				Signature: cmtrand.Bytes(MaxSignatureSize),
+			},
+			errors.New("CompactBlock: Height cannot be negative"),
+		},
+		// {
```

Comment:
should we add a todo to fix this test? or could be deleted?

Reply:
yeah I have a todo somewhere to fix the validate basic when we fix the verification